raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate_regression.py

```diff
@@ -429,3 +432,132 @@ def test_regression_mediator_task_no_routes():
+
+
+def test_regression_mediator_not_update_payer_state_twice():
+    """ Regression Test for https://github.com/raiden-network/raiden/issues/3086
+    Make sure that after a lock expired the mediator doesn't update the pair
+    twice causing EventUnlockClaimFailed to be generated at every block.
+    """
+    amount = 10
+    block_number = 5
+    initiator = HOP1
+    initiator_key = HOP1_KEY
+    mediator_address = HOP2
+    target = HOP3
+    expiration = 30
+    pseudo_random_generator = random.Random()
+
+    payer_channel = factories.make_channel(
+        partner_balance=amount,
+        our_balance=amount,
+        # partner_address=UNIT_TRANSFER_SENDER,
+        our_address=mediator_address,
+        partner_address=initiator,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    payer_route = factories.route_from_channel(payer_channel)
+
+    payer_transfer = factories.make_signed_transfer_for(
+        channel_state=payer_channel,
+        amount=amount,
+        initiator=initiator,
+        target=target,
+        expiration=expiration,
+        secret=UNIT_SECRET,
+        sender=initiator,
+        pkey=initiator_key,
+    )
+
+    payee_channel = factories.make_channel(
+        our_balance=amount,
+        our_address=mediator_address,
+        partner_address=target,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    available_routes = [factories.route_from_channel(payee_channel)]
+    channel_map = {
+        payee_channel.identifier: payee_channel,
+        payer_channel.identifier: payer_channel,
+    }
+
+    init_state_change = ActionInitMediator(
+        available_routes,
+        payer_route,
+        payer_transfer,
+    )
+    initial_state = None
+    iteration = mediator.state_transition(
+        initial_state,
+        init_state_change,
+        channel_map,
+        pseudo_random_generator,
+        block_number,
+    )
+    assert iteration.new_state is not None
+
+    current_state = iteration.new_state
+    send_transfer = must_contain_entry(iteration.events, SendLockedTransfer, {})
+    assert send_transfer
+
+    transfer = send_transfer.transfer
+    block_expiration_number = transfer.lock.expiration + DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS * 2
+
+    block = Block(
+        block_number=block_expiration_number,
+        gas_limit=1,
+        block_hash=factories.make_transaction_hash(),
+    )
+    iteration = mediator.state_transition(
+        current_state,
+        block,
+        channel_map,
+        pseudo_random_generator,
+        block_expiration_number,
+    )
+
+    msg = 'At the expiration block we should get an EventUnlockClaimFailed'
+    assert must_contain_entry(iteration.events, EventUnlockClaimFailed, {}), msg
+
+    current_state = iteration.new_state
+    block = Block(
+        block_number=block_expiration_number + 1,
+        gas_limit=1,
+        block_hash=factories.make_transaction_hash(),
+    )
+
+    receive_secret = ReceiveSecretReveal(
```

Comment:
Since the placement of this line is key can you add a comment here:

```python
# Initiator receives the secret reveal after the lock expired
```

Reply:
Done